utils.py

```diff
@@ -0,0 +1,95 @@
+from models import Item, Annotator, db, NoResultFound
+from functools import wraps
+import csv
+import io
+import crowd_bt
+
+from numpy.random import choice, random, shuffle
+from flask import Response, request, session
+from settings import ADMIN_PASSWORD, ANNOTATOR_ID
+
+def check_auth(username, password):
+    return username == 'admin' and password == ADMIN_PASSWORD
+
+def authenticate():
+    return Response('Access denied.', 401,
+        {'WWW-Authenticate': 'Basic realm="Login Required"'})
+
+def requires_auth(f):
+    @wraps(f)
+    def decorated(*args, **kwargs):
+        auth = request.authorization
+        if not auth or not check_auth(auth.username, auth.password):
+            return authenticate()
+        return f(*args, **kwargs)
+    return decorated
+
+def maybe_init_annotator(annotator):
+    if annotator.next is None:
+        # XXX this is inefficient, better to do exclude in a query
+        # ignored_ids = map(lambda i: i.id, annotator.ignore)
+        # items = filter(lambda i: i.id not in ignored_ids, Item.query.all())
+        ignored_ids = {i.id for i in annotator.ignore}
+        items = [i for i in Item.query.all() if i.id not in ignored_ids]
```

Comment:
Nice change (yay, more efficient code)!

Could you just delete those commented lines above? Or was there any particular reason they are there?